main/src/cgeo/geocaching/ui/EditButton.java

```diff
@@ -0,0 +1,152 @@
+package cgeo.geocaching.ui;
+
+import android.content.Context;
+import android.text.Editable;
+import android.text.InputType;
+import android.text.TextWatcher;
+import android.util.AttributeSet;
+import android.view.Gravity;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.RelativeLayout;
+import android.widget.Toast;
+
+import cgeo.geocaching.R;
+
+import static android.text.InputType.TYPE_TEXT_FLAG_CAP_CHARACTERS;
+import static android.text.InputType.TYPE_TEXT_FLAG_NO_SUGGESTIONS;
+
+/**
+ * This class allows for a user to change the text on a button with a long-click.
+ */
+
+public class EditButton extends RelativeLayout {
+
+    EditText edit;  // EditText used to facilitate keyboard entry.
+    Button butt;    // The actual button used for the most part.
+
+    private class CoordDigitLongClickListener implements View.OnLongClickListener {
+
+        // This implementation was obtained from 'Adithya' via the stack overflow question 'Long-press on Button to Change Text':
+        // https://stackoverflow.com/questions/44858720/long-press-on-button-to-change-text/44859328#44859328
+
+        @Override
+        public boolean onLongClick(final View view) {
+
+            final InputMethodManager imm = (InputMethodManager)
+                    getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
+
+            butt.setVisibility(View.INVISIBLE);
+            edit.setVisibility(View.VISIBLE);
+
+            if (edit.requestFocus()) {
+                imm.showSoftInput(edit, InputMethodManager.SHOW_IMPLICIT);
+            }
+
+            edit.addTextChangedListener(new TextWatcher() {
```

Comment:
That probably would be neater.  I'll look into it.
It's now down at line 133 by the way ;).